nibabel/affines.py

```diff
@@ -296,3 +295,31 @@ def voxel_sizes(affine):
     """
     top_left = affine[:-1, :-1]
     return np.sqrt(np.sum(top_left ** 2, axis=0))
+
+
+def obliquity(affine):
+    r"""
+    Estimate the *obliquity* an affine's axes represent.
+
+    The term *obliquity* is defined here as the rotation of those axes with
+    respect to the cardinal axes.
+    This implementation is inspired by `AFNI's implementation
+    <https://github.com/afni/afni/blob/b6a9f7a21c1f3231ff09efbd861f8975ad48e525/src/thd_coords.c#L660-L698>`_.
+    For further details about *obliquity*, check `AFNI's documentation
+    <https://sscc.nimh.nih.gov/sscc/dglen/Obliquity>_.
+
+    Parameters
+    ----------
+    affine : 2D array-like
+        Affine transformation array.  Usually shape (4, 4), but can be any 2D
+        array.
+
+    Returns
+    -------
+    angles : 1D array-like
+        The *obliquity* of each axis with respect to the cardinal axes, in radians.
+
+    """
+    vs = voxel_sizes(affine)
+    best_cosines = np.abs((affine[:-1, :-1] / vs).max(axis=1))
```

Comment:
Oh, sorry, I meant:

```
best_cosines = np.abs(affine[:-1, :-1] / vs).max(axis=1)
```

Don't you then need to take the min?    As in:

```
return np.arccos(best_cosines.min())
```

?  This will give the arccos of the largest deviation ...  I think.

Reply:
Sorry - by 'largest deviation' I mean the one with the angle furthest from zero.

Reply:
Now the function returns the obliquity of the three axes, if that makes sense - up to the user to pick the most deviated.

Reply:
What do you think about `best_cosines = np.abs(affine[:-1, :-1] / vs).max(axis=1)` ?

Reply:
Good catch, thanks!